qt/aqt/webview.py

```diff
@@ -28,35 +28,88 @@
 if TYPE_CHECKING:
     from aqt.mediasrv import PageContext
 
+BridgeCommandHandler = Callable[[str], Any]
 
-# Page for debug messages
-##########################################################################
 
-BridgeCommandHandler = Callable[[str], Any]
+class AnkiWebViewKind(Enum):
+    """Enum registry of all web views managed by Anki
+
+    The value of each entry corresponds to the web view's title.
+
+    When introducing a new web view, please add it to the registry below.
+    """
+
+    DEFAULT = "default"
+    MAIN = "main webview"
+    TOP_TOOLBAR = "top toolbar"
+    BOTTOM_TOOLBAR = "bottom toolbar"
+    DECK_OPTIONS = "deck options"
+    EDITOR = "editor"
+    LEGACY_DECK_STATS = "legacy deck stats"
+    DECK_STATS = "deck stats"
+    PREVIEWER = "previewer"
+    CHANGE_NOTETYPE = "change notetype"
+    CARD_LAYOUT = "card layout"
+    BROWSER_CARD_INFO = "browser card info"
+    IMPORT_CSV = "csv import"
+    EMPTY_CARDS = "empty cards"
+    FIND_DUPLICATES = "find duplicates"
+    FIELDS = "fields"
+    IMPORT_LOG = "import log"
+    IMPORT_ANKI_PACKAGE = "anki package import"
+
+
+class AuthInterceptor(QWebEngineUrlRequestInterceptor):
+    _api_enabled = False
+
+    def __init__(self, parent: QObject | None = None, api_enabled: bool = False):
+        super().__init__(parent)
+        self._api_enabled = api_enabled
+
+    def interceptRequest(self, info):
+        from aqt.mediasrv import _APIKEY
+
+        if self._api_enabled:
+            info.setHttpHeader(b"Authorization", f"Bearer {_APIKEY}".encode("utf-8"))
+
+
+_profile_with_api_access: QWebEngineProfile | None = None
+_profile_without_api_access: QWebEngineProfile | None = None
 
 
 class AnkiWebPage(QWebEnginePage):
-    def __init__(self, onBridgeCmd: BridgeCommandHandler) -> None:
-        QWebEnginePage.__init__(self)
+    def __init__(
+        self,
+        onBridgeCmd: BridgeCommandHandler,
+        kind: AnkiWebViewKind,
+        parent: QObject | None,
+    ) -> None:
+        profile = self._profileForPage(kind)
+        QWebEnginePage.__init__(self, profile, parent)
         self._onBridgeCmd = onBridgeCmd
+        self._kind = kind
         self._setupBridge()
         self.open_links_externally = True
 
-    def _setupBridge(self) -> None:
-        class Bridge(QObject):
-            def __init__(self, bridge_handler: Callable[[str], Any]) -> None:
-                super().__init__()
-                self.onCmd = bridge_handler
+    def _profileForPage(self, kind: AnkiWebViewKind) -> QWebEngineProfile:
+        have_api_access = kind in (
+            AnkiWebViewKind.DECK_OPTIONS,
```

Comment:
```suggestion
            AnkiWebViewKind.MAIN,
            AnkiWebViewKind.DECK_OPTIONS,
```
The congrats screen fails to load otherwise:

![image](https://github.com/user-attachments/assets/e62ddfc2-8b13-479b-995b-ad9074834a72)

Reply:
> AnkiWebViewKind.DECK_OPTIONS,

`IMPORT_LOG`, `IMPORT_CSV` and `IMPORT_ANKI_PACKAGE` should be whitelisted as well, for their respective import dialogs to work